mypy/test/data/parse-errors.test

```diff
@@ -294,6 +294,20 @@ def f(x, y: Z): # type: (x, y) -> z
 file: note: In function "f":
 file:1: error: Function has duplicate type signatures
 
+[case testTooManyTypes]
+def f(x, y): # type: (x, y, z) -> z
```

Comment:
Maybe `x`, `y`, and `z` here should be real types?  Otherwise we rely on the implementation detail that type signature checking happens before name resolution.